progress/README.md

```diff
@@ -0,0 +1,256 @@
+# Progress
+
+## **Requests**
+
+|                  Route                   |                         Description                          |
+| :--------------------------------------: | :----------------------------------------------------------: |
+|        [GET /progress](#get-post)        | Retrieves progress entries based on the provided parameters. |
+|     [POST /progress](#post-progress)     |                Creates a new progress entry.                 |
+| [DELETE /progress/:id](#delete-progress) |       Deletes a progress entry with the specified ID.        |
+
+## **GET /progress**
+
+Retrieves progress entries based on the provided parameters.
+
+- **Params**
+  None
+- **Query**  
+  type (query parameter): Specifies the type of progress to filter (e.g., "task", "user").
+  id (query parameter): Specifies the ID of the progress entry to retrieve (optional).
+- **Body**  
+  None
+- **Headers**  
+  Content-Type: application/json
+- **Cookie**  
+  None
+- **Success Response:**
+  - **Code:** 200
+    - **Content:** 
+    ```json
+		{
+      "message": "User progress retrieved successfully.",
+      "progress": [
+        { "<user_progress_object_1>" },
+        { "<user_progress_object_2>" },
+        ..
+      ]
+    }
+    ```
+- **Error Response:**
+  - **Code:** 500
+    - **Content:** `{ 'statusCode': 500, 'error': 'Internal Server Error', 'message': 'An internal server error occurred' }`
+
+- **Example:**
+  GET /progress?type=task
+    ```json
+    {
+			"message": "All User Progress found successfully.",
+			"progress": [
+				{
+				"id": "progress1",
+				"type": "task",
+				"userId": "b0ad77cf461633s7745d59d8",
+				"taskId": "d576s3cd4450df7698731b7a",
+				"completed": "added modal progress",
+				"planned": "write test for modal progress",
+				"blockers": "None",
+				"createdAt": 1654101900000
+				},
+				{
+				"id": "progress2",
+				"type": "user",
+				"userId": "4d973377d1ad504b8cs6f576",
+				"completed": "added controllers",
+				"planned": "add model functions",
+				"blockers": "Pending review",
+				"date": "2023-04-30",
+				"createdAt": 1654101900400
+				}
+			]
+    }
+    ```
+	GET /progress?type=user&id=user123
+	```json
+	{
+		"message": "User progress retrieved successfully.",
+		"progress": [
+			{
+				"id": "user123",
+				"type": "user",
+				"userId": "user123",
+				"completed": "made backend changes for the feature a23",
+				"planned": "make frontend changes for the feature a23",
+				"blockers": "Pending review",
+				"date": "2023-04-30",
+				"createdAt": "2023-04-30T12:00:00Z"
+			}
+		]
+	}
+
+	```
+
+
+
+## **POST /progress**
```

Comment:
let's add a type to the param as it will make the API more readable since it is doing multiple tasks.

Reply:
can be POST `/progress/${id}?type="USER"`

Reply:
Will make it easier to remember as well since we will be using same API with post and get

Reply:
then in that case these both params should not be included in the payload. Right ?

Reply:
Yes, we can get users' IDs as the post will be an authenticated route. Auth function will give us their id. in case we miss this. i can take your id and fill in your progress from my end (if it is to be passed in payload)

Reply:
According to REST Principles we should use request body for required data to create a resource. Hence passing all the required data in the payload will be better. it will even make our api clean and concise.